editor/src/components/canvas/dom-walker.ts

```diff
@@ -316,12 +328,28 @@ export function useDomWalker(props: CanvasContainerProps): React.Ref<HTMLDivElem
             return `index-${index}`
           }
           const pathElement = Utils.defaultIfNullLazy<id>(uidAttribute, makeIndexElement)
-          const uniquePath: TemplatePath = TP.appendToPath(uniqueParentPath, pathElement)
-
-          let originalPath: TemplatePath | null = null
           const originalPathElement = Utils.defaultIfNull(uidAttribute, originalUIDAttribute)
-          if (originalPathElement != null && originalParentPath != null) {
-            originalPath = TP.appendToPath(originalParentPath, originalPathElement)
+          const parentAttribute = Utils.defaultIfNull(
+            parentUIDsAttribute,
+            originalParentUIDsAttribute,
+          )
+
+          // Build the unique path for this element.
+          let uniquePath: TemplatePath = uniqueParentPath
+          if (parentUIDsAttribute != null) {
+            uniquePath = TP.appendToPath(uniquePath, parentUIDsAttribute.split('/'))
+          }
+          uniquePath = TP.appendToPath(uniquePath, pathElement)
+
+          // Build the original path for this element.
```

Comment:
At this point would it make more sense to just store the full (stringified) original path on the element itself via something like `data-utopia-original-path`?

Reply:
I don't _think_ we can necessarily construct that in our `patchedCreateReactElement` function, as it wont know necessarily enough of the chain above it.

Reply:
Ah yeah, of course. Urgh.